MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolume2.h

```diff
@@ -0,0 +1,121 @@
+/** @file MockFirmwareVolume2.h
+  This file declares a mock of Firmware Volume Protocol.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#ifndef MOCK_FIRMWARE_VOLUME2_H
+#define MOCK_FIRMWARE_VOLUME2_H
+
+#include <Library/GoogleTestLib.h>
+#include <Library/FunctionMockLib.h>
+
+extern "C" {
+  #include <Uefi.h>
+  #include <Pi/PiFirmwareVolume.h>
+  #include <Pi/PiFirmwareFile.h>
+  #include <Protocol/FirmwareVolume2.h>
+}
+
+struct MockFirmwareVolume2Protocol {
+  MOCK_INTERFACE_DECLARATION (MockFirmwareVolume2Protocol);
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    GetVolumeAttributes,
+    (
+     IN CONST  EFI_FIRMWARE_VOLUME2_PROTOCOL *This,
+     OUT       EFI_FV_ATTRIBUTES             *FvAttributes
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    SetVolumeAttributes,
+    (
+     IN CONST  EFI_FIRMWARE_VOLUME2_PROTOCOL *This,
+     IN OUT    EFI_FV_ATTRIBUTES             *FvAttributes
+    )
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    FV_ReadFile,

```

Comment:
Was there a reason to change the naming convention for some of these?

Reply:
Can you link to that function please?

Reply:
Hi @makubacki,
The function is a windows api:
https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-readfile
The error message in build log as below:
![image](https://github.com/user-attachments/assets/d6c40936-50d5-4bad-9174-b72d6982b9b6)

thanks,
TF

Reply:
It might also be helpful to capture the conflict in the PR description as well, but the main concern was the naming convention. People tend to copy/paste and use code as an example and we want examples that follow the conventions.